platform/lib/routers/pages.js

```diff
@@ -144,16 +148,40 @@ if (config.environment === 'development') {
 }
 
 if (config.environment !== 'development') {
-  pages.use('/', (request, response, next) => {
-    // Check if this request should be filtered
-    const activeFormat = getFilteredFormat(request);
-    if (activeFormat) {
-      // And if it should be filtered rewrite to the correct file
-      request.url = request.url.replace('.html', `.${activeFormat}.html`);
+  const STATIC_PAGES_PATH = utils.project.absolute('platform/pages');
+  const NOT_FOUND_ROUTE = '/404.html';
+
+  pages.use('/', (request, response) => {
```

Comment:
Afaik, I you implement a middleware, you should call `next`. Or am I missing anything?

Reply:
I am unsure about this to be honest - the built-in static middleware seems to terminate requests and to directly send the file regardless any following middlewares. But we don't need to adopt this behaviour for ours, then we'd just need to set the response body without directly sending it. Do you know how?

Reply:
You could wrap the static middleware:

```
const staticMiddleware = app.static('public');
app.use((req, res, next) => {
   if (hasFilter(...)) {
      // do filter
   } else {
        staticMiddleware(req, res, next);
   }
});
```

I've used a similar pattern to enable the optimizer: https://github.com/ampproject/docs/pull/1385/files#diff-c0ef79f636352ecae379454e61514ef8R63